app/src/test/java/com/example/ohthmhyh/HabitUnitTest.java

```diff
@@ -120,6 +120,45 @@ public void Test_getters_setters() {
         assertTrue(h.getIsPrivate());
     }
 
+
+//    @Test
+//    public void test_adherance () {
+//        //test initialization
+//        Habit h = new Habit();
+//
+//
+//    }
+
+    @Test
+    public void test_Opportunity() {
+        Habit h = new Habit();
+        //h.setStartDate(1636538265); //Nov 10, 2021 (WED)
+
+        ArrayList<Habit.Days> sched = new ArrayList<Habit.Days>();
+        sched.add(Habit.Days.Mon);
+        sched.add(Habit.Days.Fri);
+        h.setSchedule(sched);
+
+        LocalDate startDay = LocalDate.of(2021, 10, 13);
+        LocalDate currentDay = LocalDate.of(2021, 11, 17);
+
+        //LocalDate startDay = LocalDate.ofEpochDay(h.getStartDate());
+        int totalOpportunity = h.calculateOpportunity(startDay, currentDay);
+        assertEquals(2, totalOpportunity);
+
+
+
+
+
+        //test sundays
+
+        //test edge case: if today is wednesday, what if it is applicable to wednesday?
+
+
+
+
+    }
+
```

Comment:
Can we get more unit tests for this opportunity calculation?

Try and think of all the edge cases and create separate test methods for each.

Reply:
Yeah we need many more lol. That's what I'm working on now